application/src/test/java/org/opentripplanner/model/TimetableSnapshotTest.java

```diff
@@ -261,6 +272,44 @@ public void testPurge() {
     assertFalse(resolver.isDirty());
   }
 
+  @Test
+  void testUniqueDirtyTimetablesAfterMultipleUpdates() {
+    TimetableSnapshot snapshot = new TimetableSnapshot();
+    TripPattern pattern = patternIndex.get(new FeedScopedId(feedId, "1.1"));
+    Trip trip = pattern.scheduledTripsAsStream().findFirst().orElseThrow();
+
+    TripTimes updatedTriptimes = TripTimesFactory.tripTimes(
+      trip,
+      List.of(new StopTime()),
+      new Deduplicator()
+    );
+    RealTimeTripUpdate realTimeTripUpdate = new RealTimeTripUpdate(
+      pattern,
+      updatedTriptimes,
+      SERVICE_DATE,
+      TripOnServiceDate.of(trip.getId()).withTrip(trip).withServiceDate(SERVICE_DATE).build(),
+      true,
+      true
+    );
+
+    snapshot.update(realTimeTripUpdate);
+    snapshot.update(realTimeTripUpdate);
+    assertTrue(snapshot.isDirty());
+
+    TransitLayerUpdater transitLayer = new TransitLayerUpdater(null) {
+      @Override
+      public void update(
```

Comment:
Can you add a check that this method is called at all? A simple (atomic) boolean is enough for me.

Reply:
Done.